tests/src/Test/Tasty/Clash.hs

```diff
@@ -98,13 +98,15 @@ data Verilate = SimAndVerilate | VerilateOnly | SimOnly
 
 data Sim = Vivado | GHDL | ModelSim | Verilator deriving Eq
 
+data Load = LoadAll | LoadNone | Bypass
+
 data TestOptions =
   TestOptions
     { hdlSim :: [Sim]
     -- ^ Run hdl simulators (GHDL, ModelSim, etc.)
-    , hdlLoad :: Bool
-    -- ^ Load hdl into simulator (GHDL, ModelSim, etc.). Disabling this will
-    -- disable 'hdlSim' too.
+    , hdlLoad :: Load
+    -- ^ Load hdl into simulator (GHDL, ModelSim, etc.). 'LoadNone' will disable
+    -- 'hdlSim' but 'Bypass' will disable loading but allow simulators
```

Comment:
It would! But none of our tests have `Bypass` except the FIFO, which sets `hdlSim = [Vivado]`.

Reply:
Makes sense. It's a good idea to document this just to reduce the "wtf per second" rate while reading the testsuite's code :-)

Reply:
Wouldn't it make more sense to make this `[Sim]` as well? You could specify which tools to test the building with and which the simulation. Also, it seems this only works for Vivado anyway as GHDL, Icarus Verilog, ModelSim and Verilator all require the `hdlLoad` phase before they can run the `hdlSim` phase. Instead of this `Bypass`, you could pass `hdlLoad = []`.

Reply:
Perhaps we should lift this bit into a new PR where we also fix the issues found after merging PR #2257.

Reply:
Agree! Sorry about tripping that up :grimacing: